### PR TITLE
adds PromptInName and Cohere queries

### PR DESCRIPTION
**Pull request by kpister**
@jjhenkel a non-zero amount of the updates here are from formatting. Do you have a preferred .prettierrc or something of that sort that you want to add to the repo? I have zero preference, except for consistency.

still to add:

- [x] openai/anthropic full completion matching
- [x] lists of dictionaries matching
- [ ] ~Langchain Tools~
- [x] Langchain Prompt or Message classes
- [ ] string evaluator

### CODE REVIEW

**Comment by jjhenkel** [comment]
No preference for me either, feel free to bring in a `.prettierrc` or an equivalent!

**Comment by kpister** [comment]
@dhia-e-rzig @jjhenkel Not sure how we want to run reviews. You can give this a look when you get a chance.

The code is mostly ported over from promptset now. Test with `yarn test`, you can see examples of prompts extracted for each of the test cases in their respective python files.

Last piece to port is the string interpreter, which will be trickier since I was using `eval` in python to simplify.

File: src/test/extension.test.ts
Comment (dhia-e-rzig) [inline src/test/extension.test.ts:1]: Is the test failing or causing an issue?

**Review by dhia-e-rzig — APPROVED** [review]
Looks good to me for the most part, only question I have is about the commented-out test. I confer the last word to @jjhenkel since he wrote the original version of the module though.

File: src/test/extension.test.ts
Comment (kpister) [reply src/test/extension.test.ts:1]: That one was failing. We can turn it back on now though.

Comment (dhia-e-rzig) [reply src/test/extension.test.ts:1]: It's probably due to missing config on your machine ( apikey  / endpoint for ChatGPT). If you could double check that would really help me debug the code! It seems to be working on my end with the latest codebase.

Just as an FYI, running the test in debug mode seems to be the only way to get the extension to work correctly. Using the regular run test button runs the extension on a stable copy of the vscode, which disables the proposed APIs.

**Comment by jjhenkel** [comment]
Hey all, just getting to this! Excited to merge this in: is it ready for merge?

**Comment by kpister** [comment]
Yep! I haven't added the metadata for each technique yet but that can be a separate PR.

**Pull request merged**